tokio/src/io/driver/interest.rs

```diff
@@ -14,6 +14,16 @@ use std::ops;
 pub struct Interest(mio::Interest);
 
 impl Interest {
+    cfg_aio! {
+        /// Interest for POSIX AIO
+        pub const AIO: Interest = Interest(mio::Interest::AIO);
+    }
+
+    cfg_aio! {
+        /// Interest for POSIX AIO lio_listio events
+        pub const LIO: Interest = Interest(mio::Interest::LIO);
+    }
```

Comment:
```suggestion
    cfg_aio! {
        /// Interest for POSIX AIO
        #[cfg(target_os = "freebsd")]
        pub const AIO: Interest = Interest(mio::Interest::AIO);

        /// Interest for POSIX AIO
        #[cfg(not(target_os = "freebsd"))]
        pub const AIO: Interest = Interest(mio::Interest::READABLE);
    }

    cfg_aio! {
        /// Interest for POSIX AIO lio_listio events
        #[cfg(target_os = "freebsd")]
        pub const LIO: Interest = Interest(mio::Interest::LIO);

        #[cfg(not(target_os = "freebsd"))]
        /// Interest for POSIX AIO lio_listio events
        pub const LIO: Interest = Interest(mio::Interest::READABLE);
    }
```

Reply:
Instead of hacking the macro to make PollAio show up in Linux's docs, how about telling docs.rs to build separate docs for FreeBSD, like this?
```diff
diff --git a/tokio/Cargo.toml b/tokio/Cargo.toml
index 3e38c0b8..241df7f1 100644
--- a/tokio/Cargo.toml
+++ b/tokio/Cargo.toml
@@ -142,6 +142,13 @@ autocfg = "1" # Needed for conditionally enabling `track-caller`
 [package.metadata.docs.rs]
 all-features = true
 rustdoc-args = ["--cfg", "docsrs"]
+targets = [
+  "i686-unknown-linux-gnu",
+  "x86_64-apple-darwin",
+  "x86_64-unknown-freebsd",
+  "x86_64-unknown-linux-gnu",
+]
+
 
 [package.metadata.playground]
 features = ["full", "test-util"]
```

Reply:
The disadvantage of that is that then `PollAio` doesn't show up if you search for it using the built-in docs.rs search bar. We do something similar for windows-specific features.

Reply:
Ok, I did it your way.  Over on Nix, I do it the other way.  Nix has far too many target_os combinations to do it your way.

Reply:
We've generally had bad experiences regarding the discoverability of our windows features, which is why we switched to doing it this way.